swarm.go

```diff
@@ -156,6 +165,9 @@ func NewSwarm(local peer.ID, peers peerstore.Peerstore, opts ...Option) (*Swarm,
 			return nil, err
 		}
 	}
+	if s.rcmgr == nil {
+		s.rcmgr = network.NullResourceManager
+	}
```

Comment:
I am a little concerned about this, if we forget the option we'll get a neutered swarm that does no resource management.

Reply:
It's a valid thing to have a swarm without resource management.
The only reason the transports are using a function parameter instead of an option is that we want to run constructor magic on them.

Reply:
ok, fair enough.